.github/workflows/scale-test-100-gce.yaml

```diff
@@ -32,6 +32,8 @@ concurrency:
 env:
   # renovate: datasource=golang-version depName=go
   go_version: 1.22.2
+  # renovate: datasource=git-refs depName=https://github.com/kubernetes/perf-tests
+  cl2_version: a7c33443c28bbecb53584e3ca26422b3938e5ff2
```

Comment:
This change is still not enough, because it is missing the definition of the target branch, and it matching the incorrect regex. You can fix it with the following patch:

```diff
diff --git a/.github/renovate.json5 b/.github/renovate.json5
index a7069b29e497..a4a133b97905 100644
--- a/.github/renovate.json5
+++ b/.github/renovate.json5
@@ -452,7 +452,8 @@
         "# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)\\s+.+kernel: '(?<currentValue>.*)'",
         '# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*)\\s+.+kernel: "(?<currentValue>.*)@(?<currentDigest>sha256:[a-f0-9]+)"',
         '# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*)\\s+.+image-version: "(?<currentValue>.*)@(?<currentDigest>sha256:[a-f0-9]+)"',
-        '# renovate: datasource=(?<datasource>.*?)\\s+.+kube-image: "(?<depName>.*):(?<currentValue>.*)@(?<currentDigest>sha256:[a-f0-9]+)"'
+        '# renovate: datasource=(?<datasource>.*?)\\s+.+kube-image: "(?<depName>.*):(?<currentValue>.*)@(?<currentDigest>sha256:[a-f0-9]+)"',
+        '# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?) branch=(?<currentValue>.*?)\\s+.+_ref: (?<currentDigest>.*)'
       ]
     },
     {
diff --git a/.github/workflows/scale-test-100-gce.yaml b/.github/workflows/scale-test-100-gce.yaml
index ed92e0d5482e..9c664e3b44b3 100644
--- a/.github/workflows/scale-test-100-gce.yaml
+++ b/.github/workflows/scale-test-100-gce.yaml
@@ -32,8 +32,8 @@ concurrency:
 env:
   # renovate: datasource=golang-version depName=go
   go_version: 1.22.2
-  # renovate: datasource=git-refs depName=https://github.com/kubernetes/perf-tests
-  cl2_version: a7c33443c28bbecb53584e3ca26422b3938e5ff2
+  # renovate: datasource=git-refs depName=https://github.com/kubernetes/perf-tests branch=master
+  cl2_ref: a7c33443c28bbecb53584e3ca26422b3938e5ff2
   # Adding k8s.local to the end makes kops happy-
   # has stricter DNS naming requirements.
   test_name: scale-100
@@ -116,7 +116,7 @@ jobs:
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
           repository: kubernetes/perf-tests
-          ref: ${{ env.cl2_version }}
+          ref: ${{ env.cl2_ref }}
           persist-credentials: false
           sparse-checkout: clusterloader2
           path: perf-tests
diff --git a/.github/workflows/scale-test-node-throughput-gce.yaml b/.github/workflows/scale-test-node-throughput-gce.yaml
index 1b409f40a592..60b32a58fac8 100644
--- a/.github/workflows/scale-test-node-throughput-gce.yaml
+++ b/.github/workflows/scale-test-node-throughput-gce.yaml
@@ -37,8 +37,8 @@ concurrency:
 env:
   # renovate: datasource=golang-version depName=go
   go_version: 1.22.2
-  # renovate: datasource=git-refs depName=https://github.com/kubernetes/perf-tests
-  cl2_version: a7c33443c28bbecb53584e3ca26422b3938e5ff2
+  # renovate: datasource=git-refs depName=https://github.com/kubernetes/perf-tests branch=master
+  cl2_ref: a7c33443c28bbecb53584e3ca26422b3938e5ff2
   # Adding k8s.local to the end makes kops happy-
   # has stricter DNS naming requirements.
   test_name: node-throughput
@@ -121,7 +121,7 @@ jobs:
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
           repository: kubernetes/perf-tests
-          ref: ${{ env.cl2_version }}
+          ref: ${{ env.cl2_ref }}
           persist-credentials: false
           sparse-checkout: clusterloader2
           path: perf-tests

```

Note that I've changed `cl2_version` into `cl2_ref` to prevent the first regex from matching it as well (as it is looking for `.+_version`). Feel free to switch the suffix to something else if you prefer, everything should be fine as long as it is not matched by any other regex (it would likely still work, but also cause more noise due to the incorrect detection).

You can verify that it actually works by running (after changing one of the refs to an older version):

```
docker run -ti -e LOG_LEVEL=debug -e GITHUB_COM_TOKEN="$(gh auth token)" \
    -v /tmp:/tmp -v $(pwd):/usr/src/app \
    docker.io/renovate/renovate:full renovate --platform=local | tee renovate.log
```

As the logs will contain something along the lines of:

```
{
  "depName": "https://github.com/kubernetes/perf-tests",
  "currentValue": "master",
  "currentDigest": "82036a7e9bdb803b5b133c2d83cfec710819f9ab",
  "datasource": "git-refs",
  "replaceString": "# renovate: datasource=git-refs depName=https://github.com/kubernetes/perf-tests branch=master\n  cl2_ref: 82036a7e9bdb803b5b133c2d83cfec710819f9ab",
  "updates": [
    {
      "updateType": "digest",
      "newValue": "master",
      "newDigest": "a7c33443c28bbecb53584e3ca26422b3938e5ff2",
      "branchName": "renovate/all-github-action"
    }
  ],
  "packageName": "https://github.com/kubernetes/perf-tests",
  "versioning": "semver-coerced",
  "warnings": []
}
```

Reply:
I would consider dropping renovate - not sure if we want to have perf-tests updated constantly. It will make it harder for use to know if the performance difference comes from updating perf-tests or some change in cilium.

Reply:
Ah yeah that's a good point @marseel. Thanks for the breakdown though @giorio94. I remember testing this locally and thinking it worked, but I don't think I looked for the JSON snippet you posted that had the `updates` key. TIL thank you!